src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
+  }
+
+  componentDidMount() {
+    const { socket, provider } = this.props;
+
+    socket.on(provider, (user) => {
+      this.popup.close();
+      this.setState({ user });
+    });
+  }
+
+  checkPopup() {
+    const check = setInterval(() => {
+      const { popup } = this;
+      if (!popup || popup.closed || popup.closed === undefined) {
+        clearInterval(check);
+        this.setState({ disabled: '' });
+      }
+    }, 1000);
+  }
+
+  openPopup() {
+    const { provider, socket } = this.props;
+    const width = 600;
+    const height = 600;
```

Comment:
Is this going to scale to all screen sizes?

Reply:
Probably not, I'll see if there's a more size-agnostic setting

Reply:
Update on this?